File: app/src/main/java/com/example/aqqhome/network/RetrofitClient.java
package com.example.aqqhome.network;
import com.example.aqqhome.utils.utils;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static Retrofit retrofit;

    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(utils.url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }


}

File: app/src/main/java/com/example/aqqhome/utils/RetrofitClient.kt
package com.example.aqqhome.utils

import com.example.aqqhome.network.ApiAQQHome
import com.example.aqqhome.utils.utils.url2
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    private const val BASE_URL = url2

    val instance: ApiAQQHome by lazy {
        val retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        retrofit.create(ApiAQQHome::class.java)
    }
}
